### ISSUE TITLE
Can't enable it in the latest VSCode 1.95.0

### ISSUE THREAD

**Issue by floyd-li** [robb0wen/synthwave-vscode#333]
Seems the latest vscode breaks something....

You may need to run VS code with admin privileges in order to enable Neon Dreams.
![CleanShot 2024-10-30 at 09 30 53@2x](https://github.com/user-attachments/assets/a439200a-c668-4460-953a-82ec7f13057d)

**Comment by NathyVZM** [comment]
Same issue here, I tried doing exactly that (run VSCode with admin privileges) and nothing, still happening 🫠 waiting for a knight in shining armor to find a solution for this

**Comment by keyork** [comment]
I'm having the same issue and the problem is coming again...

**Comment by kevinRT19** [comment]
estamos iguales caballeros apenas llego la actualización de vs code  dejo de funcionar y intente parchar el código y ni así debe ser ya un error de estructura de código o no se..

**Comment by VanXodus305** [comment]
> Same issue here, I tried doing exactly that (run VSCode with admin privileges) and nothing, still happening 🫠 waiting for a knight in shining armor to find a solution for this

I have faced the same problem even after it had been apparently fixed in #327
I did the following and the issue was fixed for me:

1) Open VS Code (Non-Administrator Mode)
2) Open the `extension.js` file in the editor from `%USERPROFILE%\.vscode\extensions\robbowen.synthwave-vscode-0.1.16\src` folder (for Windows).
3) Replace the contents of `extension.js` with the code provided below:
Courtesy of vinayakkulkarni@bd8444b05d68aca4882b5827fdff5376884faa22 (#318)
4) Save the file and restart VS Code in Administrator Mode
5) Press `Ctrl+Shift+P` and use the `Synthwave '84: Enable Neon Dreams` command.
6) Restart VS Code once again if prompted.

This should hopefully fix the issue in VS Code 1.95.0 (in Windows)
<details><summary>
extension.js (Click to expand)
</summary>
<p>

```javascript
const path = require("path");
const fs = require("fs");
const vscode = require("vscode");

/**
 * @param {vscode.ExtensionContext} context
 */
function activate(context) {
  this.extensionName = "RobbOwen.synthwave-vscode";
  this.cntx = context;

  const config = vscode.workspace.getConfiguration("synthwave84");

  let disableGlow = config && config.disableGlow ? !!config.disableGlow : false;

  let brightness =
    parseFloat(config.brightness) > 1 ? 1 : parseFloat(config.brightness);
  brightness = brightness < 0 ? 0 : brightness;
  brightness = isNaN(brightness) ? 0.45 : brightness;

  const parsedBrightness = Math.floor(brightness * 255)
    .toString(16)
    .toUpperCase();
  let neonBrightness = parsedBrightness;

  let disposable = vscode.commands.registerCommand(
    "synthwave84.enableNeon",
    function () {
      const isWin = /^win/.test(process.platform);
      const appDir = `${path.dirname(vscode.env.appRoot)}/app/out`;
      const base = appDir + (isWin ? "\\vs\\code" : "/vs/code");
      const electronBase = isVSCodeBelowVersion("1.70.0")
        ? "electron-browser"
        : "electron-sandbox";
      const htmlFileName = "workbench.html";
      const htmlFile =
        base +
        (isWin
          ? "\\" + electronBase + "\\workbench\\" + htmlFileName
          : "/" + electronBase + "/workbench/" + htmlFileName);
      const templateFile =
        base +
        (isWin
          ? "\\" + electronBase + "\\workbench\\neondreams.js"
          : "/" + electronBase + "/workbench/neondreams.js");

      try {
        // const version = context.globalState.get(`${context.extensionName}.version`);

        // generate production theme JS
        const chromeStyles = fs.readFileSync(
          __dirname + "/css/editor_chrome.css",
          "utf-8"
        );
        const jsTemplate = fs.readFileSync(
          __dirname + "/js/theme_template.js",
          "utf-8"
        );
        const themeWithGlow = jsTemplate.replace(
          /\[DISABLE_GLOW\]/g,
          disableGlow
        );
        const themeWithChrome = themeWithGlow.replace(
          /\[CHROME_STYLES\]/g,
          chromeStyles
        );
        const finalTheme = themeWithChrome.replace(
          /\[NEON_BRIGHTNESS\]/g,
          neonBrightness
        );
        fs.writeFileSync(templateFile, finalTheme, "utf-8");

        // modify workbench html
        const html = fs.readFileSync(htmlFile, "utf-8");

        // check if the tag is already there
        const isEnabled = html.includes("neondreams.js");

        if (!isEnabled) {
          // delete synthwave script tag if there
          let output = html.replace(
            /^.*(<!-- SYNTHWAVE 84 --><script src='neondreams.js'><\/script><!-- NEON DREAMS -->).*\n?/gm,
            ""
          );
          // add script tag
          output = html.replace(
            /\<\/html\>/g,
            `	<!-- SYNTHWAVE 84 --><script src='neondreams.js'></script><!-- NEON DREAMS -->\n`
          );
          output += "</html>";

          fs.writeFileSync(htmlFile, output, "utf-8");

          vscode.window
            .showInformationMessage(
              "Neon Dreams enabled. VS code must reload for this change to take effect. Code may display a warning that it is corrupted, this is normal. You can dismiss this message by choosing 'Don't show this again' on the notification.",
              { title: "Restart editor to complete" }
            )
            .then(function () {
              vscode.commands.executeCommand("workbench.action.reloadWindow");
            });
        } else {
          vscode.window
            .showInformationMessage(
              "Neon dreams is already enabled. Reload to refresh JS settings.",
              { title: "Restart editor to refresh settings" }
            )
            .then(function () {
              vscode.commands.executeCommand("workbench.action.reloadWindow");
            });
        }
      } catch (e) {
        if (/ENOENT|EACCES|EPERM/.test(e.code)) {
          vscode.window.showInformationMessage(
            "Neon Dreams was unable to modify the core VS code files needed to launch the extension. You may need to run VS code with admin privileges in order to enable Neon Dreams."
          );
          return;
        } else {
          vscode.window.showErrorMessage(
            "Something went wrong when starting neon dreams"
          );
          return;
        }
      }
    }
  );

  let disable = vscode.commands.registerCommand(
    "synthwave84.disableNeon",
    uninstall
  );

  context.subscriptions.push(disposable);
  context.subscriptions.push(disable);
}
exports.activate = activate;

// this method is called when your extension is deactivated
function deactivate() {
  // ...
}

function uninstall() {
  const isWin = /^win/.test(process.platform);
  const appDir = `${path.dirname(vscode.env.appRoot)}/app/out`;
  const base = appDir + (isWin ? "\\vs\\code" : "/vs/code");
  const electronBase = isVSCodeBelowVersion("1.70.0")
    ? "electron-browser"
    : "electron-sandbox";
  const htmlFileName = "workbench.html";
  const htmlFile =
    base +
    (isWin
      ? "\\" + electronBase + "\\workbench\\" + htmlFileName
      : "/" + electronBase + "/workbench/" + htmlFileName);

  // modify workbench html
  const html = fs.readFileSync(htmlFile, "utf-8");

  // check if the tag is already there
  const isEnabled = html.includes("neondreams.js");

  if (isEnabled) {
    // delete synthwave script tag if there
    let output = html.replace(
      /^.*(<!-- SYNTHWAVE 84 --><script src='neondreams.js'><\/script><!-- NEON DREAMS -->).*\n?/gm,
      ""
    );
    fs.writeFileSync(htmlFile, output, "utf-8");

    vscode.window
      .showInformationMessage(
        "Neon Dreams disabled. VS code must reload for this change to take effect",
        { title: "Restart editor to complete" }
      )
      .then(function () {
        vscode.commands.executeCommand("workbench.action.reloadWindow");
      });
  } else {
    vscode.window.showInformationMessage("Neon dreams isn't running.");
  }
}

// Returns true if the VS Code version running this extension is below the
// version specified in the 'version' parameter. Otherwise returns false.
function isVSCodeBelowVersion(version) {
  const vscodeVersion = vscode.version;
  const vscodeVersionArray = vscodeVersion.split(".");
  const versionArray = version.split(".");

  for (let i = 0; i < versionArray.length; i++) {
    if (vscodeVersionArray[i] < versionArray[i]) {
      return true;
    }
  }

  return false;
}

module.exports = {
  activate,
  deactivate,
};``` 
</p>
</details>

**Comment by GeorgeVorob** [comment]
> > Same issue here, I tried doing exactly that (run VSCode with admin privileges) and nothing, still happening 🫠 waiting for a knight in shining armor to find a solution for this
> 
> I have faced the same problem even after it had been apparently fixed by #327 I did the following and the issue was fixed for me:
> 
>     1. Open VS Code (Non-Administrator Mode).
> 
>     2. Open the `extension.js` file in the editor from `%USERPROFILE%\.vscode\extensions\robbowen.synthwave-vscode-0.1.16\src` folder (for Windows).
> 
>     3. Replace the contents of `extension.js` with the code provided below:
>        Courtesy of [vinayakkulkarni/synthwave-vscode@bd8444b](https://github.com/vinayakkulkarni/synthwave-vscode/commit/bd8444b05d68aca4882b5827fdff5376884faa22) ([fix: esm migration issue in VSCode #318](https://github.com/robb0wen/synthwave-vscode/pull/318))
> 
>     4. Save the file and restart VS Code in Administrator Mode.
> 
>     5. Press `Ctrl+Shift+P` and use the `Synthwave '84: Enable Neon Dreams` command.
> 
>     6. Restart VS Code once again if prompted.
> 
> 
> This should hopefully fix the issue in VS Code 1.95.0 (in Windows).
> extension.js (Click to expand)
> 
> ```js
> const path = require("path");
> const fs = require("fs");
> const vscode = require("vscode");
> 
> /**
>  * @param {vscode.ExtensionContext} context
>  */
> function activate(context) {
>   this.extensionName = "RobbOwen.synthwave-vscode";
>   this.cntx = context;
> 
>   const config = vscode.workspace.getConfiguration("synthwave84");
> 
>   let disableGlow = config && config.disableGlow ? !!config.disableGlow : false;
> 
>   let brightness =
>     parseFloat(config.brightness) > 1 ? 1 : parseFloat(config.brightness);
>   brightness = brightness < 0 ? 0 : brightness;
>   brightness = isNaN(brightness) ? 0.45 : brightness;
> 
>   const parsedBrightness = Math.floor(brightness * 255)
>     .toString(16)
>     .toUpperCase();
>   let neonBrightness = parsedBrightness;
> 
>   let disposable = vscode.commands.registerCommand(
>     "synthwave84.enableNeon",
>     function () {
>       const isWin = /^win/.test(process.platform);
>       const appDir = `${path.dirname(vscode.env.appRoot)}/app/out`;
>       const base = appDir + (isWin ? "\\vs\\code" : "/vs/code");
>       const electronBase = isVSCodeBelowVersion("1.70.0")
>         ? "electron-browser"
>         : "electron-sandbox";
>       const htmlFileName = "workbench.html";
>       const htmlFile =
>         base +
>         (isWin
>           ? "\\" + electronBase + "\\workbench\\" + htmlFileName
>           : "/" + electronBase + "/workbench/" + htmlFileName);
>       const templateFile =
>         base +
>         (isWin
>           ? "\\" + electronBase + "\\workbench\\neondreams.js"
>           : "/" + electronBase + "/workbench/neondreams.js");
> 
>       try {
>         // const version = context.globalState.get(`${context.extensionName}.version`);
> 
>         // generate production theme JS
>         const chromeStyles = fs.readFileSync(
>           __dirname + "/css/editor_chrome.css",
>           "utf-8"
>         );
>         const jsTemplate = fs.readFileSync(
>           __dirname + "/js/theme_template.js",
>           "utf-8"
>         );
>         const themeWithGlow = jsTemplate.replace(
>           /\[DISABLE_GLOW\]/g,
>           disableGlow
>         );
>         const themeWithChrome = themeWithGlow.replace(
>           /\[CHROME_STYLES\]/g,
>           chromeStyles
>         );
>         const finalTheme = themeWithChrome.replace(
>           /\[NEON_BRIGHTNESS\]/g,
>           neonBrightness
>         );
>         fs.writeFileSync(templateFile, finalTheme, "utf-8");
> 
>         // modify workbench html
>         const html = fs.readFileSync(htmlFile, "utf-8");
> 
>         // check if the tag is already there
>         const isEnabled = html.includes("neondreams.js");
> 
>         if (!isEnabled) {
>           // delete synthwave script tag if there
>           let output = html.replace(
>             /^.*(<script src='neondreams.js'><\/script>).*\n?/gm,
>             ""
>           );
>           // add script tag
>           output = html.replace(
>             /\<\/html\>/g,
>             `	<!-- SYNTHWAVE 84 --><script src='neondreams.js'></script><!-- NEON DREAMS -->\n`
>           );
>           output += "</html>";
> 
>           fs.writeFileSync(htmlFile, output, "utf-8");
> 
>           vscode.window
>             .showInformationMessage(
>               "Neon Dreams enabled. VS code must reload for this change to take effect. Code may display a warning that it is corrupted, this is normal. You can dismiss this message by choosing 'Don't show this again' on the notification.",
>               { title: "Restart editor to complete" }
>             )
>             .then(function () {
>               vscode.commands.executeCommand("workbench.action.reloadWindow");
>             });
>         } else {
>           vscode.window
>             .showInformationMessage(
>               "Neon dreams is already enabled. Reload to refresh JS settings.",
>               { title: "Restart editor to refresh settings" }
>             )
>             .then(function () {
>               vscode.commands.executeCommand("workbench.action.reloadWindow");
>             });
>         }
>       } catch (e) {
>         if (/ENOENT|EACCES|EPERM/.test(e.code)) {
>           vscode.window.showInformationMessage(
>             "Neon Dreams was unable to modify the core VS code files needed to launch the extension. You may need to run VS code with admin privileges in order to enable Neon Dreams."
>           );
>           return;
>         } else {
>           vscode.window.showErrorMessage(
>             "Something went wrong when starting neon dreams"
>           );
>           return;
>         }
>       }
>     }
>   );
> 
>   let disable = vscode.commands.registerCommand(
>     "synthwave84.disableNeon",
>     uninstall
>   );
> 
>   context.subscriptions.push(disposable);
>   context.subscriptions.push(disable);
> }
> exports.activate = activate;
> 
> // this method is called when your extension is deactivated
> function deactivate() {
>   // ...
> }
> 
> function uninstall() {
>   const isWin = /^win/.test(process.platform);
>   const appDir = `${path.dirname(vscode.env.appRoot)}/app/out`;
>   const base = appDir + (isWin ? "\\vs\\code" : "/vs/code");
>   const electronBase = isVSCodeBelowVersion("1.70.0")
>     ? "electron-browser"
>     : "electron-sandbox";
>   const htmlFileName = "workbench.html";
>   const htmlFile =
>     base +
>     (isWin
>       ? "\\" + electronBase + "\\workbench\\" + htmlFileName
>       : "/" + electronBase + "/workbench/" + htmlFileName);
> 
>   // modify workbench html
>   const html = fs.readFileSync(htmlFile, "utf-8");
> 
>   // check if the tag is already there
>   const isEnabled = html.includes("neondreams.js");
> 
>   if (isEnabled) {
>     // delete synthwave script tag if there
>     let output = html.replace(
>       /^.*(<script src='neondreams.js'><\/script>).*\n?/gm,
>       ""
>     );
>     fs.writeFileSync(htmlFile, output, "utf-8");
> 
>     vscode.window
>       .showInformationMessage(
>         "Neon Dreams disabled. VS code must reload for this change to take effect",
>         { title: "Restart editor to complete" }
>       )
>       .then(function () {
>         vscode.commands.executeCommand("workbench.action.reloadWindow");
>       });
>   } else {
>     vscode.window.showInformationMessage("Neon dreams isn't running.");
>   }
> }
> 
> // Returns true if the VS Code version running this extension is below the
> // version specified in the 'version' parameter. Otherwise returns false.
> function isVSCodeBelowVersion(version) {
>   const vscodeVersion = vscode.version;
>   const vscodeVersionArray = vscodeVersion.split(".");
>   const versionArray = version.split(".");
> 
>   for (let i = 0; i < versionArray.length; i++) {
>     if (vscodeVersionArray[i] < versionArray[i]) {
>       return true;
>     }
>   }
> 
>   return false;
> }
> 
> module.exports = {
>   activate,
>   deactivate,
> };``` 
> </p>
> </details>
> ```

Does not work for me. When enabling with your code:
![изображение](https://github.com/user-attachments/assets/1beaa300-f416-4975-9ef0-c39cad9da123)

**Comment by VanXodus305** [comment]
@GeorgeVorob Try uninstalling the extension and installing it again before modifying the file.

**Comment by lvandernoll** [comment]
https://github.com/robb0wen/synthwave-vscode/issues/333#issuecomment-2446244618

Thanks, worked for me on MacOS too

**Comment by kevinRT19** [comment]
intente pero esta diferente  diferente el codigo en esta ultima actualizacion alguien sabe en donde hay que modificar veo que cambiaron el codigo @VanXodus305 sabes en donde hay que modificarlo ya que veo el codigo mio diferente 











[codigo.txt](https://github.com/user-attachments/files/17573644/codigo.txt)

**Comment by NathyVZM** [comment]
Did as @VanXodus305 aaand also copied the code @GeorgeVorob sent and works like a charm, thank you very much! 💗

**Comment by NathyVZM** [comment]
> intente pero esta diferente diferente el codigo en esta ultima actualizacion alguien sabe en donde hay que modificar veo que cambiaron el codigo @VanXodus305 sabes en donde hay que modificarlo ya que veo el codigo mio diferente
> 
> [codigo.txt](https://github.com/user-attachments/files/17573644/codigo.txt)

Copia el código que envió @GeorgeVorob y reemplaza todo el archivo de `extension.js`, debería funcionar 👍🏻 en caso de que no, desinstala la extensión, reinstálala y haces lo del archivo nuevamente.

Déjanos saber si te funciona 🫂

**Comment by kevinRT19** [comment]
> > intente, pero está diferente el código en esta última actualización, alguien sabe en dónde hay que modificar, veo que cambiaron el código,  sabes en dónde hay que modificarlo, ya que veo el código mio diferente
> > [codigo.txt](https://github.com/user-attachments/files/17573644/codigo.txt)
> 
> Copia el código que envió @GeorgeVorob y reemplaza todo el archivo de , debería funcionar 👍🏻 en caso de que no, desinstala la extensión, reinstálala y haces lo del archivo nuevamente.`extension.js`
> 
> Déjanos saber si te funciona 🫂

ya lo intente borre la extension y la volvi  instalar y remplace todo el codigo pero aun me sigue saliendo este alerta 
![Captura de pantalla 2024-10-30 085217](https://github.com/user-attachments/assets/4823a2fe-66d1-4f1b-aada-ba9f67ae08bb)




_
y tambien veo que al final del codigo hay un error 
![Captura de pantalla 2024-10-30 085452](https://github.com/user-attachments/assets/26765960-a96a-4760-9ff6-b3d5b0ee34a9)
alguna solucion :( :

**Comment by kevinRT19** [comment]
> > > intente, pero está diferente el código en esta última actualización, alguien sabe en dónde hay que modificar, veo que cambiaron el código, sabes en dónde hay que modificarlo, ya que veo el código mio diferente [codigo.txt](https://github.com/user-attachments/files/17573644/codigo.txt)
> > 
> > 
> > Copia el código que envió @GeorgeVorob y reemplaza todo el archivo de , debería funcionar 👍🏻 en caso de que no, desinstala la extensión, reinstálala y haces lo del archivo nuevamente.`extension.js`
> > Déjanos saber si te funciona 🫂
> 
> ya lo intente borre la extensión y la volví instalar y remplace todo el código pero aún me sigue saliendo esta alerta ![Captura de pantalla 2024-10-30 085217](https://private-user-images.githubusercontent.com/174817336/381578889-4823a2fe-66d1-4f1b-aada-ba9f67ae08bb.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzAyOTY4MDQsIm5iZiI6MTczMDI5NjUwNCwicGF0aCI6Ii8xNzQ4MTczMzYvMzgxNTc4ODg5LTQ4MjNhMmZlLTY2ZDEtNGYxYi1hYWRhLWJhOWY2N2FlMDhiYi5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQxMDMwJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MTAzMFQxMzU1MDRaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1jNWI5MTZhN2VhNTk1Y2MwODUwMDIxMmIyNWU4ODJkYTk0YmVkN2RlMWQ5MjI0ZTgwMDllNWQ0YThiZjI1ZWE2JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.hjqf19zucL4Ie5ghbgf21NiLJJKbWwOYviZ2ClnoA8Y)
> 
> _ y también veo que al final del código hay un error ![Captura de pantalla 2024-10-30 085452](https://private-user-images.githubusercontent.com/174817336/381579265-26765960-a96a-4760-9ff6-b3d5b0ee34a9.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzAyOTY4MDQsIm5iZiI6MTczMDI5NjUwNCwicGF0aCI6Ii8xNzQ4MTczMzYvMzgxNTc5MjY1LTI2NzY1OTYwLWE5NmEtNDc2MC05ZmY2LWIzZDViMGVlMzRhOS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQxMDMwJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MTAzMFQxMzU1MDRaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT03Y2Q5MzZiNTQxNDNhY2M3MWUxMjZkY2MzMGIwOTdjZjE3ZWE4N2U1MGZjOGRjYzUxNTljNTlhZWFhZjlhZDNjJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.vvDv_mXuDpEjSQD0y5dtIcLUC9ZsbzTzc8woTIWd5ps) alguna solución :( :

ya encontre el problema  con el codigo 
[correcion del codigo.txt](https://github.com/user-attachments/files/17574031/correcion.del.codigo.txt)
aqui esta el codigo completo con la correccion por si no les llega funcionar el otro codigo como ami 👍

**Comment by VanXodus305** [comment]
Sorry for the trouble @kevinRT19.
The code had some markdown tags at the end which were not required. I've removed them and updated the comment.
Hope it works fine now 👍

**Comment by Manu254rce** [comment]
I think for people like me the solution was to uninstall vscode [1.95](https://code.visualstudio.com/updates/v1_95) and install [1.94](https://code.visualstudio.com/updates/v1_94). Just make sure you set settings sync on

Then after downgrading open Vscode on your settings, search for the option to auto update vscode, set it to none...

Or in the `settings.json` , add the line 

```json
"update.mode": "none"
``` 

The issue seems to be related with the fact that Vscode cannot resolve Synthwave 84 files as soon as it updates...

**Comment by ireznik** [comment]
Seems that the fix that was for 1.94 is no longer working with v1.95. The filename was changed back to **workbench.html** which was before **workbench.esm.html**

This is the commit which reverted the change in vscode: [revert](https://github.com/microsoft/vscode/commit/d2701aa2ec51f213293772173067b173cad98b2c)

**Comment by robb0wen** [comment]
I've just published a fix in v0.1.17, which should now be available on the VS code marketplace.

@ireznik as I mentioned in the PR, I think we were both working on this at the same time. Thanks again for the effort and I'm sorry I wasn't able to merge your PR instead
